ivy/functional/ivy/experimental/linear_algebra.py

```diff
@@ -1666,3 +1666,61 @@ def dot(
     ivy.array([[-15.28]])
     """
     return current_backend(a, b).dot(a, b, out=out)
+
+
+@handle_nestable
+@to_native_arrays_and_back
+@handle_exceptions
+def general_inner_product(
+    a: Union[ivy.Array, ivy.NativeArray],
+    b: Union[ivy.Array, ivy.NativeArray],
+    n_modes: Optional[int] = None,
+) -> Union[ivy.Array, float]:
+    """
+    Generalised inner products between tensors.
+
+        Takes the inner product between the last (respectively first)
+        `n_modes` of `a` (respectively `b`)
+
+    Parameters
+    ----------
+    a, b : tensor
+    n_modes : int, default is None
+        * if None, the traditional inner product is returned (i.e. a float)
+        * otherwise, the product between the `n_modes` last modes of `a`
+            and the `n_modes` first modes of `b` is returned.
+            The resulting tensor's order is `ndim(a) - n_modes`.
```

Comment:
Sorry for nitpicking here but the standard parameter definition in ivy is:
```
<parameter_name>
     <parameter_description>
```

Would you please make this change everywhere?